modules/gdscript/gdscript.cpp

```diff
@@ -2043,20 +2043,26 @@ void GDScriptInstance::notification(int p_notification, bool p_reversed) {
 	//notification is not virtual, it gets called at ALL levels just like in C.
 	Variant value = p_notification;
 	const Variant *args[1] = { &value };
+	const StringName &notification_str = GDScriptLanguage::get_singleton()->strings._notification;
 
-	List<GDScript *> pl;
-	GDScript *sptr = script.ptr();
-	while (sptr) {
-		if (p_reversed) {
-			pl.push_back(sptr);
+	static thread_local Vector<GDScript *> script_stack;
+	int script_count = 0;
+	for (GDScript *sptr = script.ptr(); sptr; sptr = sptr->_base, ++script_count) {
+		if (unlikely(script_stack.size() == script_count)) {
+			script_stack.push_back(sptr);
 		} else {
-			pl.push_front(sptr);
+			script_stack.set(script_count, sptr);
 		}
-		sptr = sptr->_base;
 	}
-	for (GDScript *sc : pl) {
+
+	const int start = p_reversed ? 0 : script_count - 1;
+	const int end = p_reversed ? script_count : -1;
+	const int step = p_reversed ? 1 : -1;
+
+	for (int idx = start; idx != end; idx += step) {
```

Comment:
I'd use a "lower than" check just to be safe (if I am understanding this right that is)
```suggestion
	for (int idx = start; idx < end; idx += step) {
```

Reply:
Note that `step` can be either positive or negative.

Reply:
Ah I see. I did not realise. Given that the step is only ever a whole step of 1, this _shouldn't_ be an issue...